qtpy/tests/test_qtgui.py

```diff
@@ -1,22 +1,47 @@
 """Test QtGui."""
+import os
+import sys
+
 import pytest
 
 from qtpy import PYQT5, PYQT_VERSION, QtGui
 
 
-def test_qdrag_functions():
+@pytest.mark.skipif(sys.platform.startswith('linux') and os.environ.get('USE_CONDA', 'No') == 'No',
+                    reason="Fatal Python error: Aborted on Linux CI when not using conda")
+def test_qfontmetrics_width(qtbot):
+    """Test QFontMetrics width"""
+    assert QtGui.QFontMetrics.width is not None
+    font = QtGui.QFont("times", 24)
+    font_metrics = QtGui.QFontMetrics(font)
+    width = font_metrics.width("Test")
+    assert width in range(40, 62)
+
+
+@pytest.mark.skipif(sys.platform.startswith('linux') and os.environ.get('USE_CONDA', 'No') == 'No',
+                    reason="Fatal Python error: Aborted on Linux CI when not using conda")
+def test_qdrag_functions(qtbot):
     """Test functions mapping for QtGui.QDrag."""
-    assert QtGui.QDrag.exec_
+    assert QtGui.QDrag.exec_ is not None
+    drag = QtGui.QDrag(None)
+    drag.exec_()
 
 
 def test_qguiapplication_functions():
     """Test functions mapping for QtGui.QGuiApplication."""
-    assert QtGui.QGuiApplication.exec_
+    assert QtGui.QGuiApplication.exec_ is not None
 
 
+@pytest.mark.skipif(sys.platform.startswith('linux') and os.environ.get('USE_CONDA', 'No') == 'No',
+                    reason="Segmentation fault/Aborted on Linux CI when not using conda")
 def test_qtextdocument_functions():
     """Test functions mapping for QtGui.QTextDocument."""
-    assert QtGui.QTextDocument.print_
+    assert QtGui.QTextDocument.print_ is not None
+    text_document = QtGui.QTextDocument("Test")
+    print_device = QtGui.QPdfWriter('test.pdf')
+    text_document.print_(print_device)
+    assert os.path.exists('test.pdf')
+    os.remove('test.pdf')
```

Comment:
This file will not get cleaned up if the test fails, potentially interfering with any other tests that use it. To avoid this, use a yield fixture and request it from the test. Furthermore, since the same setup and teardown code is used in multiple tests, encapsulating it into a fixture inside `conftest.py` is more DRY and allows re-using it across tests.

Also, given we support Python 3.6 at minimum, its much nicer to use `pathlib`. 

```python
@pytest.fixture
def pdf_writer():
    output_path = Path('test.pdf')
    device = QtGui.QPdfWriter(str(output_path))
    yield device, output_path
    output_path.unlink()